controllers/nodehealthcheck_controller.go

```diff
@@ -476,7 +476,7 @@ func (r *NodeHealthCheckReconciler) deleteOrphanedRemediationCRs(nhc *remediatio
 
 		// check if node exists
 		for _, node := range allNodes {
-			if node.GetName() == cr.GetName() {
+			if node.GetName() == utils.GetNodeNameFromCR(cr) {
```

Comment:
fixes `[FAIL] Node Health Check CR Reconciliation with expected permanent node deletion [It] it should delete orphaned CR when node is deleted`